src/OutputReaders/field_time_series_indexing.jl

```diff
@@ -227,6 +229,18 @@ end
 ##### FieldTimeSeries updating
 #####
 
+# Let's make sure `times` is available on the CPU. This is always the case 
+# for ranges. if `times` is a vector that resides on the GPU, it has to be moved to the CPU for safe indexing. 
+# TODO: Copying the whole array is a bit unclean, maybe find a way that avoids the penalty of allocating and copying memory.
+# This would require refactoring `FieldTimeSeries` to include a cpu-allocated times array
+cpu_available_interpolating_time_indices(::CPU, times, time_indexing, t, arch) = interpolating_time_indices(time_indexing, times, t)
+cpu_available_interpolating_time_indices(::CPU, times::AbstractVector, time_indexing, t) = interpolating_time_indices(time_indexing, times, t)
+
+function cpu_available_interpolating_time_indices(::GPU, times::AbstractVector, time_indexing, t) 
+    cpu_times = on_architecture(CPU(), times)
+    return interpolating_time_indices(time_indexing, cpu_times, t)
```

Comment:
Would it be better to launch a small kernel, eg something like

```julia
ñ = Ref(1.0) # ?
n₁ = Ref(1)
n₂ = Ref(1)

# launch kernel that stores indices in `ñ, n₁, n₂`
```

Reply:
I had tried that, but it seems to not work with `Ref`, allocating a `CuArray` to hold an index seems a bit much. I am not sure if there is a type that allows exchanging data from CPU to GPU inside a kernel

Reply:
I think that's what is done when doing a global reduction over all dimensions, eg

https://github.com/JuliaGPU/GPUArrays.jl/blob/cd1f59ae2fd17763ca73505f966b980bf7135354/src/host/mapreduce.jl#L71